lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2868,6 +2841,116 @@ ParseResult FIRStmtParser::parseStaticRefExp(Value &result,
+  SymbolValueEntry symtabEntry;
+  if (parseId(id, message) ||
+      moduleContext.lookupSymbolEntry(symtabEntry, id, loc))
+    return failure();
+
+  Type type;
+  if (auto *op = symtabEntry.dyn_cast<Operation *>()) {
+    // It is targeting the operation itself, not a port.
+    result = hw::InnerSymTarget(op);
+    return success();
+  } else {
+    // This target can be a port or a regular value.
+    auto value = symtabEntry.get<Value>();
+    type = value.getType();
+    if (isa<OpResult>(value)) {
+      // This target is a regular value, such as a wire or node.
+      result = hw::InnerSymTarget(value.getDefiningOp());
+    } else {
+      // This target is a module port.
+      auto blockArg = cast<BlockArgument>(value);
+      auto *module = blockArg.getOwner()->getParentOp();
+      auto portNo = blockArg.getArgNumber();
+      result = hw::InnerSymTarget(portNo, module);
+    }
+  }
+
+  // We have our target operation, we just need to parse the field id.
+  uint64_t fieldID = 0;
+  while (true) {
+    if (consumeIf(FIRToken::period)) {
+      SmallVector<StringRef, 3> fields;
+      if (parseFieldIdSeq(fields, "expected field name"))
+        return failure();
+      for (auto fieldName : fields) {
+        if (auto bundle = type_dyn_cast<BundleType>(type)) {
+          if (auto index = bundle.getElementIndex(fieldName)) {
+            fieldID += bundle.getFieldID(*index);
+            continue;
+          }
+        } else if (auto bundle = type_dyn_cast<OpenBundleType>(type)) {
+          if (auto index = bundle.getElementIndex(fieldName)) {
+            fieldID += bundle.getFieldID(*index);
+            continue;
+          }
+        } else {
+          return emitError(loc, "subfield requires bundle operand ");
+        }
+        return emitError(loc,
+                         "unknown field '" + fieldName + "' in bundle type ")
+               << type;
+      }
+    } else if (consumeIf(FIRToken::l_square)) {
+      auto loc = getToken().getLoc();
+      int32_t index;
+      if (parseIntLit(index, "expected index") ||
+          parseToken(FIRToken::r_square, "expected ']'"))
+        return failure();
+
+      if (index < 0)
+        return emitError(loc, "invalid index specifier"), failure();
+
+      if (auto vector = type_dyn_cast<FVectorType>(type)) {
+        if ((unsigned)index < vector.getNumElements()) {
+          fieldID += vector.getFieldID(index);
+          continue;
+        }
+      } else if (auto vector = type_dyn_cast<FVectorType>(type)) {
+        if ((unsigned)index < vector.getNumElements()) {
+          fieldID += vector.getFieldID(index);
+          continue;
+        }
+      } else {
+        return emitError(loc, "subfield requires bundle operand ");
+      }
+      return emitError(loc, "invalid index specifier");
+    } else {
+      break;
+    }
+  }
+  result = hw::InnerSymTarget::getTargetForSubfield(result, fieldID);
+  return success();
+}
+
+/// path ::= 'path(' inner_sym_target ')'
```

Comment:
Works for me, but the grammar rule here probably shouldn't be so specific to inner symbols/target, I think this could be `static_reference` ?

Reply:
We already have a rule for `static_reference` and `parseStaticReference`, but as you know, its the same idea.  I named this after the function since its result type is an InnerSymTarget, which, admittedly, still sucks.  I think/hope we can combine them in to a single function in a future PR, so it should be temporary.  Any suggestion for a different name? Or should I just use `static_reference` here and drop the grammar comment on `parseInnerSymTarget`?

Reply:
Alright, I just inlined this function is PathOp, after some review it doesn't seem exactly reusable for the probe operations.  For the grammar I just changed it to use `static_reference`.